src/internal.c

```diff
@@ -6829,7 +6829,22 @@ int SetSSL_CTX(WOLFSSL* ssl, WOLFSSL_CTX* ctx, int writeDup)
     ssl->buffers.certChainCnt = ctx->certChainCnt;
 #endif
 #ifndef WOLFSSL_BLIND_PRIVATE_KEY
+#ifdef WOLFSSL_COPY_KEY
+    if (ctx->privateKey != NULL) {
+        if (ssl->buffers.key != NULL) {
+            FreeDer(&ssl->buffers.key);
```

Comment:
I think if free'ing a previous key here we need a check for ssl->buffers.weOwnKey and follow the same pattern as wolfSSL_UnloadCertsKeys().

Reply:
I actually disagree. Unlike wolfSSL_UnloadCertsKeys() we are with certainty going to set the ssl->buffers.key. Checking on weOwnCert as well just introduces the chance that through mismanagement if weOwnCert is incorrectly not set at the time of this check and we leak memory. This flag is also partially to ensure behavior replication for openssl compat. Regardless of the state of weOwnCert, if this flag is enabled we should always be copying the key material in my opinion as that is what openssl does.

Reply:
Biggest concern was a case state where the pointer is pointing to the key from the WOLFSSL_CTX. I don't think that state could be hit here though when looking more. We do not check before assigning for the blinding case.